https://bugs.scummvm.org/ticket/14262

SDL 1.2.15 did some "OS X 10.6 SDK deprecation cleanup" and added a restriction
in SDL_Init() on OSX, giving an "Unsupported display mode!" initialization error,
when one doesn't have their monitor configured to millions of colors.

I can't see why this restriction was added for older OSX. It used to work just
fine, and it still appears to be OK when removing that check.

See more original context in:
https://github.com/libsdl-org/SDL-1.2/commit/991bbb91c8cff091c7c1aee216e02949080756b6

--- a/src/video/quartz/SDL_QuartzVideo.m
+++ b/src/video/quartz/SDL_QuartzVideo.m
@@ -288,10 +288,23 @@
     }
 #endif
 
+    /*
+     * XXX: SDL 1.2.15 added this restriction to 32-bit modes, but on
+     * OSX 10.4 it's easy to have the monitor settings set to thousands
+     * (i.e. 16bpp) instead of millions of colors, especially when
+     * using Classic or playing other games which will lower this
+     * setting.  And, more importantly, not having this restriction
+     * worked fined on older OSX, so revert this SDL 1.2.15 change.
+     *
+     * This avoids having ScummVM fail early in SDL_Init() with
+     * "Unsupported display mode!".
+     */
+#if 0
     /* we only care about the 32-bit modes... */
     if (*bpp != 32) {
         *bpp = 0;
     }
+#endif
 }
 
 static int QZ_VideoInit (_THIS, SDL_PixelFormat *video_format)
